### PR TITLE
Build 8.1.1 release candidate

### PR DESCRIPTION
**Pull request by peastman**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)

### CODE REVIEW

**Comment by peastman** [comment]
@conda-forge-admin, please rerender

**Comment by peastman** [comment]
@jakirkham I merged this PR, but there's no sign of it building the packages.  I did an empty push to `rc` to try to force it, but still nothing.

**Comment by jakirkham** [comment]
Weird. Not sure why that happened. Manually started [a new CI build]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=853152&view=results )

**Comment by peastman** [comment]
Thanks!  Unfortunately, all the builds are failing when trying to solve the environment.

```
The reported errors are:
- Encountered problems while solving:
-   - nothing provides requested cuda-nvcc_linux-64 11.8.*
```

File: .ci_support/linux_64_c_compiler_version11cuda_compilernvcccuda_compiler_version11.8cxx_compiler_version11numpy1.22python3.10.____cpython.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+cdt_name:
+- cos7
+channel_sources:
+- conda-forge
+channel_targets:
+- conda-forge openmm_rc
+cuda_compiler:
+- nvcc
+cuda_compiler_version:
+- '11.8'
```
Comment (jakirkham) [inline .ci_support/linux_64_c_compiler_version11cuda_compilernvcccuda_compiler_version11.8cxx_compiler_version11numpy1.22python3.10.____cpython.yaml:14]: If I look at [this CI build]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=853152&view=logs&jobId=3a25022a-f08c-5c04-be7d-d279a7fc48ef&j=d1321064-f6c3-56b7-0172-6d994d01c836&t=39022612-2533-571e-7b41-94814e7d38bb ), it points to this variant file

Looking at the `cuda_compiler` above it show it as `nvcc`, which it should as it is CUDA 11 (not `cuda-nvcc` as used by CUDA 12 or in the error message). This appears to be the case for the rest of the variants as well

Is something else setting `cuda_compiler` later? Could we try re-rendering with the latest changes again?

File: .ci_support/linux_64_c_compiler_version11cuda_compilernvcccuda_compiler_version11.8cxx_compiler_version11numpy1.26python3.12.____cpython.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+cdt_name:
+- cos7
+channel_sources:
+- conda-forge
+channel_targets:
+- conda-forge openmm_rc
+cuda_compiler:
+- nvcc
+cuda_compiler_version:
+- '11.8'
```
Comment (jakirkham) [inline .ci_support/linux_64_c_compiler_version11cuda_compilernvcccuda_compiler_version11.8cxx_compiler_version11numpy1.26python3.12.____cpython.yaml:14]: Interestingly [this CI build passes]( linux_64_c_compiler_version11cuda_compilernvcccuda_compiler_version11.8cxx_compiler_version11numpy1.26python3.12.____cpython.yaml ) though the values are exactly the same as the previous variant (both as would be expected for CUDA 11.8)

File: .ci_support/linux_64_c_compiler_version11cuda_compilernvcccuda_compiler_version11.8cxx_compiler_version11numpy1.22python3.10.____cpython.yaml (hunk 1 above)
Comment (peastman) [reply .ci_support/linux_64_c_compiler_version11cuda_compilernvcccuda_compiler_version11.8cxx_compiler_version11numpy1.22python3.10.____cpython.yaml:14]: How do I do that?  Just create a new empty PR and tell it to rerender?

**Comment by jakirkham** [comment]
Thanks for letting me know Peter! 🙏

There are certainly a lot of failures that look rather odd, but it also appears there are quite a few builds that pass

What is particularly unusual is the failures are mostly with Linux `x86_64` on CUDA 11.8 where `cuda_compiler` appears to be set as `nvcc`, which should result in using `nvcc_linux-64`, but instead `cuda-nvcc_linux-64`. Am not sure where that setting is coming from if not the variant files. Have noted a failing and passing build for Linux `x86_64` on CUDA 11.8 above. Both set `cuda_compiler` to `nvcc` so both shouldn't see the issue (yet they do)

Somewhere `cuda_compiler` is changing to `cuda-nvcc`. Not entirely sure where that would be if not the variant files

**Comment by jakirkham** [comment]
cc @jaimergp (in case you have any insights on what is happening here)

**Comment by peastman** [comment]
Possibly I made a mistake in manually resolving all the conflicts between the main and rc branches?  There were a lot of them, and it took quite a long time.  But the build passed before I merged the PR.  (Question for @jaimergp: is it really necessary to have separate main and rc branches?  It means a painful manual merge every time we switch between them.)

Since those are all generated files, can we just tell it to regenerate them without changing anything else?

**Comment by isuruf** [comment]
Yes, it was an issue with your manual merge and push. This is why we require everything to be done in PRs instead of pushing to branches manually.

**Comment by isuruf** [comment]
No, you pushed f132dc2cb82fc4806640f0365e058effbf0b2b35 without a PR

**Comment by isuruf** [comment]
That got commit was the problematic one

**Comment by peastman** [comment]
That's because it failed to build when I merged the PR.  The instructions @jaimergp wrote up said that if it fails to build, the solution is to force it with a direct push to the branch.  If there's a better way of doing it, I'd love to know what it is.

**Comment by isuruf** [comment]
Only use a empty commit and push if the builds don't start. Do not make any functional changes outside of a PR.

**Comment by peastman** [comment]
I didn't make any functional changes outside the PR.  My intent was to use an empty commit.  But like I said above, every time we switch branches we end up with a horrible manual merge that is very difficult to do correctly.  That's why I'm asking whether there's a way to avoid having to do that?

**Pull request merged**